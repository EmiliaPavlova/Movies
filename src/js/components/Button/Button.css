.button {
  padding: 0;
  border: none;
  font-weight: bold;
  font-size: 12px;
  font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
  background: none;
  cursor: pointer;
}

.button:focus {
  outline: 0;
}
