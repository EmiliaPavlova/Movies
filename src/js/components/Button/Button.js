import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import './Button.css';

const Button = ({
  title,
  className,
  selected,
  onClick,
  ...rest
}) => {
  const classNames = `button ${className ? className : ''} ${selected ? 'selected' : ''}`;

  return (
    <button className={classNames} onClick={onClick} {...rest}>
      {title}
    </button>
  )
}

Button.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  selected: PropTypes.bool,
  onClick: PropTypes.func.isRequired
};

export default Button;
