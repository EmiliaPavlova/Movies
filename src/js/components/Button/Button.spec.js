import React from 'react';
import { mount } from 'enzyme';
import Button from './Button';
import './Button.css';

const componentProps = (overrides = {}) => ({
  title: 'Button',
  className: 'class',
  selected: false,
  onClick: jest.fn(),
  ...overrides
});

const mountButton = props => mount(<Button {...props} />);

describe('Button', () => {
  describe('Button attributes', () => {
    it('matches the snapshot', () => {
      const wrapper = mountButton(componentProps());
      expect(wrapper).toMatchSnapshot();
    });
    it('sets a custom class when provided via `className` property', () => {
      const props = componentProps({ className: 'custom-class' });
      const wrapper = mountButton(props);
      const button = wrapper.find('button').at(0);
      expect(button.props().className).toBe(`button custom-class `);
    });
    it('sets a \'selected\' class when button is selected', () => {
      const props = componentProps({ selected: true });
      const wrapper = mountButton(props);
      const button = wrapper.find('button').at(0);
      expect(button.props().className).toBe(`button class selected`);
    });
  });
  describe('Button events', () => {
    it('sets the onClick event handler with the given function via the `onClick` property', () => {
      const onClickMock = jest.fn();
      const props = componentProps({ onClick: onClickMock });
      const wrapper = mountButton(props);
      const button = wrapper.find('button').at(0);

      expect(button.props().onClick).toBe(onClickMock);

      button.simulate('click');
      expect(onClickMock).toHaveBeenCalledTimes(1);
      expect(onClickMock).toHaveBeenCalledWith(expect.any(Object));
    });

  })
});
