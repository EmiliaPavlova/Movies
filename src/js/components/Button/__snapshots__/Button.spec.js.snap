// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button attributes matches the snapshot 1`] = `
<Button
  className="class"
  onClick={[MockFunction]}
  selected={false}
  title="Button"
>
  <button
    className="button class "
    onClick={[MockFunction]}
  >
    Button
  </button>
</Button>
`;
